src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/controller/v1/FlowController.java

```diff
@@ -328,18 +332,39 @@ public void invalidateFlowCache() {
      */
     @ApiOperation(value = "Gets history for flow", response = FlowEventPayload.class, responseContainer = "List")
     @GetMapping(path = "/{flow_id}/history")
-    @ResponseStatus(HttpStatus.OK)
-    public CompletableFuture<List<FlowEventPayload>> getHistory(
+    public ResponseEntity<List<FlowEventPayload>> getHistory(
             @PathVariable("flow_id") String flowId,
-            @ApiParam(value = "default: the day before timeTo.", required = false)
+            @ApiParam(value = "default: 0 (1 January 1970 00:00:00).", required = false)
             @RequestParam(value = "timeFrom", required = false) Optional<Long> optionalTimeFrom,
             @ApiParam(value = "default: now.", required = false)
-            @RequestParam(value = "timeTo", required = false) Optional<Long> optionalTimeTo) {
+            @RequestParam(value = "timeTo", required = false) Optional<Long> optionalTimeTo,
+            @ApiParam(value = "Return at most N latest records. "
+                    + "Default: if `timeFrom` or/and `timeTo` parameters are presented default value of "
+                    + "`maxCount` is infinite (all records in time interval will be returned). "
+                    + "Otherwise default value of `maxCount` will be equal to 100. In This case response will contain "
+                    + "header 'Content-Range'.")
+            @RequestParam(value = "max_count", required = false) Optional<Integer> optionalMaxCount) {
+        int maxCount = optionalMaxCount.orElseGet(() -> {
+            if (optionalTimeFrom.isPresent() || optionalTimeTo.isPresent()) {
+                return Integer.MAX_VALUE;
+            } else {
+                return DEFAULT_MAX_HISTORY_RECORD_COUNT;
+            }
+        });
+
         Long timeTo = optionalTimeTo.orElseGet(() -> Instant.now().getEpochSecond());
-        Long timeFrom = optionalTimeFrom.orElseGet(
-                () -> Instant.ofEpochSecond(timeTo).minus(1, ChronoUnit.DAYS).getEpochSecond()
-        );
-        return flowService.listFlowEvents(flowId, timeFrom, timeTo);
+        Long timeFrom = optionalTimeFrom.orElse(0L);
+        List<FlowEventPayload> events = flowService.listFlowEvents(flowId, timeFrom, timeTo, maxCount).join();
+
+        HttpHeaders headers = new HttpHeaders();
+
+        if (!optionalMaxCount.isPresent() && !optionalTimeFrom.isPresent() && !optionalTimeTo.isPresent()) {
+            // if request has no parameters we assume that default value of `maxCount` is 100. To indicate that response
+            // may contain not all of history records "Content-Range" header will be added to response.
+            int count = Math.min(DEFAULT_MAX_HISTORY_RECORD_COUNT, events.size()) - 1;
+            headers.add(HttpHeaders.CONTENT_RANGE, format("items 0-%d/*", count));
```

Comment:
I'd return this header ONLY in cases when we have in database more records than DEFAULT_MAX_HISTORY_RECORD_COUNT.